app/services/foreman_discovery/host_converter.rb

```diff
@@ -21,10 +21,9 @@ def self.to_managed(original_host, set_managed = true, set_build = true, added_a
   def self.set_build_clean_facts(host)
     # fact cleaning
     if Setting['discovery_clean_facts']
-      # clean all facts except those starting with "discovery_"
+      # clean all facts except those from Discovery
       host.define_singleton_method(:clear_facts) do
-          keep_ids = FactValue.where("host_id = #{host.id}").joins(:fact_name).where("fact_names.name like 'discovery_%'").pluck("fact_values.id")
-          FactValue.where("host_id = #{host.id} and id not in (?)", keep_ids).delete_all
+          FactValue.where.not(host_id: FactValue.where(host_id: host.id, fact_name: { type: 'DiscoveryFactName' }).joins(:fact_name).pluck(:id)).delete_all
```

Comment:
You should be able to omit the pluck here, see

https://github.com/rubocop-hq/rubocop-rails/issues/246

Reply:
Nit: Since there are so many methods involved, it will be difficult to debug in the future if anything went wrong. Lets break it to getting the `id` in one and then deleting it in the next. What do you think @domitea?

Reply:
Thanks @timogoebel for tip what to omit.

I understand, that huge one liner can be difficult to debug @rabajaj0509. So I split that into two lines.